error: failed to run custom build command for `neptune-triton v0.2.2`

Compiling ocl-core-vector v0.1.0
   Compiling tokio-tcp v0.1.4
   Compiling tempfile v3.1.0
   Compiling libsecp256k1 v0.3.5
   Compiling tls-api v0.3.2
The following warnings were emitted during compilation:

warning: ./lib/a.c:14:10: fatal error: CL/cl.h: No such file or directory
warning:  #include <CL/cl.h>
warning:           ^~~~~~~~~
warning: compilation terminated.

error: failed to run custom build command for `neptune-triton v0.2.2`

Caused by:
  process didn't exit successfully: `/home/sw/workspace/filecoin/forest/target/release/build/neptune-triton-f8e585453ad11c46/build-script-build` (exit code: 1)
--- stdout
TARGET = Some("x86_64-unknown-linux-gnu")
OPT_LEVEL = Some("3")
HOST = Some("x86_64-unknown-linux-gnu")
CC_x86_64-unknown-linux-gnu = None
CC_x86_64_unknown_linux_gnu = None
HOST_CC = None
CC = None
CFLAGS_x86_64-unknown-linux-gnu = None
CFLAGS_x86_64_unknown_linux_gnu = None
HOST_CFLAGS = None
CFLAGS = None
CRATE_CC_NO_DEFAULTS = None
DEBUG = Some("false")
CARGO_CFG_TARGET_FEATURE = Some("fxsr,mmx,sse,sse2")
running: "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-shared" "-Wall" "-Wextra" "-fPIC" "-lOpenCL" "-o" "/home/sw/workspace/filecoin/forest/target/release/build/neptune-triton-532c42a02dd1b116/out/./lib/a.o" "-c" "./lib/a.c"
cargo:warning=./lib/a.c:14:10: fatal error: CL/cl.h: No such file or directory
cargo:warning= #include <CL/cl.h>
cargo:warning=          ^~~~~~~~~
cargo:warning=compilation terminated.
exit code: 1

--- stderr


error occurred: Command "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-shared" "-Wall" "-Wextra" "-fPIC" "-lOpenCL" "-o" "/home/sw/workspace/filecoin/forest/target/release/build/neptune-triton-532c42a02dd1b116/out/./lib/a.o" "-c" "./lib/a.c" with args "cc" did not execute successfully (status code exit code: 1).



warning: build failed, waiting for other jobs to finish...
error: failed to compile `forest v0.1.0 (/home/sw/workspace/filecoin/forest/forest)`, intermediate artifacts can be found at `/home/sw/workspace/filecoin/forest/target`

Caused by:
  build failed
Makefile:4: recipe for target 'install' failed
make: *** [install] Error 101

--- a/.github/PULL_REQUEST_TEMPLATE.md
+++ b/.github/PULL_REQUEST_TEMPLATE.md
@@ -5,8 +5,9 @@ Changes introduced in this pull request:
 
 
 **Reference issue to close (if applicable)**
-<!-- Include the issue reference this pull request is connected to (e.g. closes #1). -->
-Closes <!--ADD ISSUE NUMBER (don't remove Closes)-->
+<!-- Include the issue reference this pull request is connected to -->
+<!--(e.g. Closes #1)-->
+Closes 
 
 
 **Other information and links**

--- a/README.md
+++ b/README.md
@@ -39,7 +39,7 @@ make install
 forest
 ```
 
-> protoc is also required to build, if not already installed follow [instructions here](http://google.github.io/proto-lens/installing-protoc.html)
+> `protoc` and `OpenCL` are also required to build. Installation for protoc can be found [here](http://google.github.io/proto-lens/installing-protoc.html).
 
 ### Config
 

--- a/forest/Cargo.toml
+++ b/forest/Cargo.toml
@@ -7,7 +7,7 @@ edition = "2018"
 [dependencies]
 forest_libp2p = { path = "../node/forest_libp2p" }
 utils = { path = "../node/utils" }
-db = { path = "../node/db" }
+db = { path = "../node/db", features = ["rocksdb"] }
 libp2p = "0.16.2"
 futures = "0.3.1"
 log = "0.4.8"
@@ -21,6 +21,6 @@ state_manager = { path = "../blockchain/state_manager" }
 cid = { package = "forest_cid", path = "../ipld/cid" }
 forest_car = { path = "../ipld/car" }
 blocks = { package = "forest_blocks", path = "../blockchain/blocks" }
-ipld_blockstore = { path = "../ipld/blockstore" }
+ipld_blockstore = { path = "../ipld/blockstore", features = ["rocksdb"] }
 chain = { path = "../blockchain/chain" }
-structopt = { version = "0.3"}
+structopt = { version = "0.3" }

--- a/ipld/blockstore/Cargo.toml
+++ b/ipld/blockstore/Cargo.toml
@@ -10,3 +10,6 @@ db = { path = "../../node/db" }
 encoding = { package = "forest_encoding", path = "../../encoding" }
 forest_ipld = { path = "../" }
 commcid = { path = "../../utils/commcid" }
+
+[features]
+rocksdb = ["db/rocksdb"]

--- a/ipld/blockstore/src/lib.rs
+++ b/ipld/blockstore/src/lib.rs
@@ -6,9 +6,12 @@ mod buffered;
 pub use self::buffered::BufferedBlockStore;
 
 use cid::{multihash::MultihashDigest, Cid};
-use db::{MemoryDB, RocksDb, Store};
+use db::{MemoryDB, Store};
 use encoding::{de::DeserializeOwned, from_slice, ser::Serialize, to_vec};
 use std::error::Error as StdError;
+
+#[cfg(feature = "rocksdb")]
+use db::RocksDb;
 
 /// Wrapper for database to handle inserting and retrieving ipld data with Cids
 pub trait BlockStore: Store {
@@ -42,4 +45,6 @@ pub trait BlockStore: Store {
 }
 
 impl BlockStore for MemoryDB {}
+
+#[cfg(feature = "rocksdb")]
 impl BlockStore for RocksDb {}

--- a/node/db/Cargo.toml
+++ b/node/db/Cargo.toml
@@ -5,7 +5,7 @@ authors = ["ChainSafe Systems <info@chainsafe.io>"]
 edition = "2018"
 
 [dependencies]
-rocksdb = "0.14.0"
+rocksdb = { version = "0.14.0", optional = true }
 parking_lot = "0.10.0"
 encoding = { package = "forest_encoding", path = "../../encoding" }
 thiserror = "1.0"

--- a/node/db/src/errors.rs
+++ b/node/db/src/errors.rs
@@ -11,6 +11,7 @@ pub enum Error {
     InvalidBulkLen,
     #[error("Cannot use unopened database")]
     Unopened,
+    #[cfg(feature = "rocksdb")]
     #[error(transparent)]
     Database(#[from] rocksdb::Error),
     #[error(transparent)]
@@ -26,6 +27,7 @@ impl PartialEq for Error {
         match (self, other) {
             (&InvalidBulkLen, &InvalidBulkLen) => true,
             (&Unopened, &Unopened) => true,
+            #[cfg(feature = "rocksdb")]
             (&Database(_), &Database(_)) => true,
             (&Encoding(_), &Encoding(_)) => true,
             (&Other(ref a), &Other(ref b)) => a == b,

--- a/node/db/src/lib.rs
+++ b/node/db/src/lib.rs
@@ -7,6 +7,8 @@ mod rocks;
 
 pub use errors::Error;
 pub use memory::MemoryDB;
+
+#[cfg(feature = "rocksdb")]
 pub use rocks::RocksDb;
 
 pub trait DatabaseService {

--- a/node/db/src/rocks.rs
+++ b/node/db/src/rocks.rs
@@ -1,6 +1,8 @@
 // Copyright 2020 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
+#![cfg(feature = "rocksdb")]
+
 use super::errors::Error;
 use super::{DatabaseService, Store};
 use rocksdb::{Options, WriteBatch, DB};

--- a/node/db/tests/db_utils/mod.rs
+++ b/node/db/tests/db_utils/mod.rs
@@ -1,6 +1,8 @@
 // Copyright 2020 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
+#![cfg(feature = "rocksdb")]
+
 // Taken from
 // https://github.com/rust-rocksdb/rust-rocksdb/blob/master/tests/util/mod.rs
 use rocksdb::{Options, DB};

--- a/node/db/tests/rocks_test.rs
+++ b/node/db/tests/rocks_test.rs
@@ -1,6 +1,8 @@
 // Copyright 2020 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
+#![cfg(feature = "rocksdb")]
+
 mod db_utils;
 mod subtests;
 

